tools/dcos_login.py

```diff
@@ -187,32 +27,78 @@ def login_session() -> None:
     DCOS_LOGIN_PASSWORD: the EE password (defaults to deleteme)
     DCOS_ENTERPRISE: determine how to authenticate (defaults to false)
     DCOS_ACS_TOKEN: bypass auth and use the user supplied token
-    """
-    cluster_url = os.environ.get("CLUSTER_URL")
-    if not cluster_url:
-        raise Exception("Must have CLUSTER_URL set in environment!")
 
+    This is the current flow to login to a cluster
+    - Dockerfile downloads the latest cli that works for 1.10 and above ONLY.
+      |- If the cluster is 1.10 and above:
+         |- If Open:
+            |- Export DCOS_ACS_TOKEN
+            |- Then do a `dcos cluster setup <url>`, it should magically work!
+         |- If Enterprise:
+            |- A `dcos cluster setup` with username and password should work.
+      |- If the cluster is less than 1.10: <WE ONLY SUPPORT 1.9 - TO BE DEPRECATED>
+         |- Overwrite the binary to 1.9
+         |- For Open AND Enterprise, use `dcos config set` for all the below properties:
+            |- core.dcos_url
+            |- core.ssl_verify
+            |- core.dcos_acs_token
+    """
     def ignore_empty(envvar, default):
         # Ignore the user passing in empty ENVVARs.
         value = os.environ.get(envvar, "").strip()
-        if not value:
-            return default
-
-        return value
+        return value if value else default
 
     dcos_login_username = ignore_empty("DCOS_LOGIN_USERNAME", __CLI_LOGIN_EE_USERNAME)
     dcos_login_password = ignore_empty("DCOS_LOGIN_PASSWORD", __CLI_LOGIN_EE_PASSWORD)
-    dcos_enterprise = ignore_empty("DCOS_ENTERPRISE", "true").lower() == "true"
-    dcos_acs_token = os.environ.get("DCOS_ACS_TOKEN")
-    if not dcos_acs_token:
-        dcos_acs_token = login(
-            dcosurl=cluster_url,
-            username=dcos_login_username,
-            password=dcos_login_password,
-            is_enterprise=dcos_enterprise,
-        )
-    configure_cli(dcosurl=cluster_url, token=dcos_acs_token)
+    if ignore_empty("DCOS_ENTERPRISE", "true").lower() == "true":
+        _run_cmd(
+            "dcos cluster setup {} --username {} --password {} --insecure".format(
+                cluster_url,
+                dcos_login_username,
+                dcos_login_password
+            ),
+            check=True)
+    else:
+        # This would try to use `xdg-open` and print a warning that it was not found in the PATH,
+        # but reads stdin and cluster will setup successfully.
+        rc, _, _ = _run_cmd(
+            "dcos cluster setup {} --insecure".format(cluster_url),
+            check=True,
+            cmd_input=bytes(os.environ.get(__DCOS_ACS_TOKEN_ENV, __CLI_LOGIN_OPEN_TOKEN), encoding="utf-8"))
+    _run_cmd("dcos node --json", check=True)
+
+
+def _run_cmd(cmd, check=False, cmd_input=None):
+    log.info("[CMD] {}".format(cmd))
```

Comment:
This might leak credentials since the command is called with `--username {}` and `--password {}`. I really would avoid this type of authentication. What's the reason not to use environment variables?